### PR TITLE
fix: disable smart swaps for Snap accounts

### PR DESCRIPTION
**Pull request by montelaidev**
## **Description**

This pr disables swaps for snap accounts.

## **Related issues**

Resolves: https://github.com/MetaMask/accounts-planning/issues/234 

## **Manual testing steps**

See before/after recordings

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/selectors/selectors.js [hunk 1]
```diff
@@ -209,6 +209,17 @@ export function isHardwareWallet(state) {
...
+/**
+ * Checks if the current account is a Snap account.
+ *
+ * @param {object} state - The state object.
+ * @returns {boolean}
+ */
+export function isAccountSnap(state) {
+  const keyring = getCurrentKeyring(state);
+  return Boolean(keyring?.type?.includes('Snap'));
+}
```
Comment (gantunesr) [inline ui/selectors/selectors.js:221]: `isAccountSnap` or `isSnapAccount`?

Comment (gantunesr) [inline ui/selectors/selectors.js:221]: Can we use a constant value here, something like,

```suggestion
export function isAccountSnap(state) {
  const keyring = getCurrentKeyring(state);
  return keyring?.type === KeyringType.snap;
}
```

Also, I see we have the selector `getAccountType`, lets make sure we're not duplicating code unnecessarily

File: ui/ducks/swaps/swaps.js [hunk 2]
```diff
@@ -323,6 +324,7 @@ export const getSmartTransactionsErrorMessageDismissed = (state) =>
 
 export const getSmartTransactionsEnabled = (state) => {
   const hardwareWalletUsed = isHardwareWallet(state);
+  const accountSnapUsed = isAccountSnap(state);
```
Comment (gantunesr) [inline ui/ducks/swaps/swaps.js:327]: Small nit: Maybe we could group this into something like `supportsSmartTx`

**Review by gantunesr — APPROVED** [review]

**Review by danroc — APPROVED** [review]

**Review by danroc — APPROVED** [review]
LGTM

**Pull request merged**